### PR TITLE
Enable `react/no-array-index-key` rule

### PR DESCRIPTION
**Pull request by fregante**
## What does this PR do?

- Part of https://github.com/pixiebrix/eslint-config-pixiebrix/issues/194
- Mentioned in https://github.com/pixiebrix/pixiebrix-extension/issues/7019#issuecomment-1932279426

## Checklist

- [x] Designate a primary reviewer: @BLoe

### CODE REVIEW

File: src/extensionConsole/pages/mods/ModsPageActions.tsx [hunk 1]
```diff
@@ -42,72 +42,47 @@ const ModsPageActions: React.FunctionComponent<{
...
-        title: (
-          <>
-            {/* Applying the same classes which <FontAwesomeIcon/> applies */}
-            <PublishIcon className="svg-inline--fa fa-w-16 fa-fw" /> Publish to
-            Marketplace
-          </>
-        ),
+        title: "Publish to Marketplace",
+        // Applying the same classes which <FontAwesomeIcon/> applies
+        icon: <PublishIcon className="svg-inline--fa fa-w-16 fa-fw" />,
```
Comment (fregante) [inline src/extensionConsole/pages/mods/ModsPageActions.tsx:47]: With hundreds of available icons, I'm confused why we need any more custom ones. Font Awesome has this: 

https://fontawesome.com/icons/arrow-up-from-bracket?f=classic&s=solid

Comment (fregante) [reply src/extensionConsole/pages/mods/ModsPageActions.tsx:47]: It looks like that's only available on v6, while we're still on v5


- https://github.com/pixiebrix/pixiebrix-extension/issues/3844

**Review by grahamlangford — APPROVED** [review]

File: src/components/ellipsisMenu/EllipsisMenu.tsx [hunk 2]
```diff
@@ -152,16 +154,16 @@ const EllipsisMenu: React.FunctionComponent<EllipsisMenuProps> = ({
...
             ) : (
               <Dropdown.Item
-                key={index}
+                key={item.title}
                 onClick={item.action}
                 className={item.className}
                 disabled={item.disabled}
               >
-                {item.title}
+                {item.icon} {item.title}
```
Comment (twschiller) [inline src/components/ellipsisMenu/EllipsisMenu.tsx:166]: Will this introduce an extra space if no icon is provided?

Comment (fregante) [reply src/components/ellipsisMenu/EllipsisMenu.tsx:166]: Technically yes, but spaces are collapsed at the beginning of block elements. That's why you can do:

```
<div>Content</div>
<div>
    More content
</div>
```

**Pull request merged**